src/reducers/stats.js

```diff
@@ -4,30 +4,28 @@ import initialPlayerStats from './initialPlayerStats';
 
 // TODO update babel etc
 
+const adjustmentsByType = {
```

Comment:
As it stands this is basically just abstracting the common behaviour that's already here, into a more data-driven form.

As you'll see, a similar pattern pops up in `reducers/statusEffectsTick`. I think the overall architecture you might want in the long-run here is along the lines of:

```High level action (e.g. "drink water") => actionToAdjustments (e.g. "drink => hydration up by drink.value, bladderFullness up by drink.fullness") => apply all adjustments to stats```

Adding that intermediate stage lets you share the plumbing for stats, tick, status tick, etc. Your actions are translated into an intermediate data structure representing the effect that needs to be applied, then there's shared code that handles "take these effects and tell me what the new stats should be".

On its own that may be a bit dry (all your status effects and in-game actions boil down to "number go up/down"), but you can add handlers for fancier stuff on top of it. This architecture would serve to give you a robust baseline for the most common case (which is, presumably, "number go up/down").